src/Infrastructure/Data/DataTransferObject.php

```diff
@@ -73,11 +73,11 @@ abstract public function toArray();
      *
      * @param array $rawData Raw DTO data.
      * @param string $key Data key.
-     * @param string $default Default value.
+     * @param mixed $default Default value.
      *
      * @return mixed
      */
-    protected static function getDataValue(array $rawData, $key, $default = '')
+    protected static function getDataValue(array $rawData, string $key, mixed $default = '')
```

Comment:
The support for [mixed type](https://www.php.net/manual/en/language.types.mixed.php) (including type declaration for function arguments) was added in PHP 8.0.0, core library supports PHP 7.2 and up. We should only keep the doc block type hinting here.

Reply:
Thanks @goran-stamenkovski-logeecom! You are right, We found it out running the tests and was fixed at https://github.com/sequra/integration-core/pull/17/files#diff-cf94c85ea73d2e2fb47007dfe9491f4dd0835754f42d2badb9f5ab7bae82ed4bR80